### PR TITLE
[WIP] Add Travis CI integration.

### PR DESCRIPTION
**Pull request by tonycosentini**

### CODE REVIEW

**Comment by spirosoik** [comment]
@tonycosentini You need to add these in your `travis.yml` because by default Travis uses `build-tools-21.1.2`

```
android:
  components:
    - build-tools-21.1.1
    - android-21
```

**Comment by jaredsburrows** [comment]
Can we get this fixed?

**Comment by tonycosentini** [comment]
I think this is going to be tricky - we need to fix both Android tools (probably easy) and also figure out how to install Infer on Travis (not sure how hard this is).

I can take a look at some point, but if someone is particularly interested - feel free to commandeer.

**Comment by jaredsburrows** [comment]
@tonycosentini `infer` works on linux and osx. Please see the docs here: http://fbinfer.com/docs/getting-started.html.

**Comment by tonycosentini** [comment]
Yup - it definitely works, just need to add the installation to the Travis file.

**Comment by vgaidarji** [comment]
@tonycosentini do you have any plans to finish Travis configuration soon? 
If yes - just perfect, if no - please let me know and I take care of it.
Thanks.

**Comment by tonycosentini** [comment]
Apoligies for letting this sit stagnant for so long, @vgaidarji feel free to take it over if you want :).

I'm not sure if @clonetwin26 or @kageiit have any feedback, but ideally the job would just run the build task (which includes some integration tests).

File: infer-sample/build.gradle [hunk 1]
```diff
@@ -10,13 +10,13 @@ dependencies {
 }
 
 android {
-    buildToolsVersion "23.0.2"
-    compileSdkVersion 23
+    buildToolsVersion "21.1.1"
```
Comment (kageiit) [inline infer-sample/build.gradle]: Can we update to latest?

File: infer-sample/build.gradle [hunk 2]
```diff
@@ -10,13 +10,13 @@ dependencies {
...
+    compileSdkVersion 21
 
     defaultConfig {
         applicationId "com.uber.inferplugin"
         minSdkVersion 15
-        targetSdkVersion 22
+        targetSdkVersion 21
```
Comment (kageiit) [inline infer-sample/build.gradle]: Latest target sdk

**Comment by vgaidarji** [comment]
Ok, thanks @tonycosentini. I'll work on it asap.

**Comment by vgaidarji** [comment]
@tonycosentini hey, could you please add me to the list of contributors of this repository, so that I have enough access rights and configure Travis fully?
Thanks.

**Comment by tonycosentini** [comment]
I can't change access for the repository, but if you fork it and have a working configuration we can enable it.

**Comment by vgaidarji** [comment]
@tonycosentini Ok, I'll work in separate branch then in my fork.

**Comment by tonycosentini** [comment]
This is now a reality thanks to contributions from @vgaidarji!

**Pull request closed without merging**